src/fns.c

```diff
@@ -979,33 +979,6 @@ If STRING is multibyte and contains a character of charset
   return string;
 }
 
-DEFUN ("string-to-unibyte", Fstring_to_unibyte, Sstring_to_unibyte,
-       1, 1, 0,
-       doc: /* Return a unibyte string with the same individual chars as STRING.
-If STRING is unibyte, the result is STRING itself.
-Otherwise it is a newly created string, with no text properties,
-where each `eight-bit' character is converted to the corresponding byte.
-If STRING contains a non-ASCII, non-`eight-bit' character,
-an error is signaled.  */)
-  (Lisp_Object string)
-{
-  CHECK_STRING (string);
-
-  if (STRING_MULTIBYTE (string))
-    {
-      ptrdiff_t chars = SCHARS (string);
-      unsigned char *str = xmalloc (chars);
-      ptrdiff_t converted = str_to_unibyte (SDATA (string), str, chars);
-
-      if (converted < chars)
-	error ("Can't convert the %"pD"dth character to unibyte", converted);
```

Comment:
It took me a while to figure out that "pD" is a macro. Based on this comment in src/lisp.h
```
/* Use pD to format ptrdiff_t values, which suffice for indexes into
   buffers and strings.  Emacs never allocates objects larger than
   PTRDIFF_MAX bytes, as they cause problems with pointer subtraction.
   In C99, pD can always be "t"; configure it here for the sake of
   pre-C99 libraries such as glibc 2.0 and Solaris 8.  */
```
It looks like we can just make our rust code use %td for ptrdiff_t's?